packages/flutter/lib/src/material/material.dart

```diff
@@ -475,3 +498,132 @@ abstract class InkFeature {
   @override
   String toString() => describeIdentity(this);
 }
+
+/// An interpolation between two [ShapeBorder]s.
+///
+/// This class specializes the interpolation of [Tween] to use [ShapeBorder.lerp].
+class ShapeBorderTween extends Tween<ShapeBorder> {
+  /// Creates a [ShapeBorder] tween.
+  ///
+  /// the [begin] and [end] properties may be null; see [ShapeBorder.lerp] for
+  /// the null handling semantics.
+  ShapeBorderTween({ShapeBorder begin, ShapeBorder end}): super(begin: begin, end: end);
+
+  /// Returns the value this tween has at the given animation clock value.
+  @override
+  ShapeBorder lerp(double t) {
+    return ShapeBorder.lerp(begin, end, t);
+  }
+}
+
+/// Animated version of [PhysicalShape].
+///
+/// The [elevation] is animated.
+///
+/// The [color] is animated if the [animateColor] property is set; otherwise,
+/// the color changes immediately at the start of the animation for the other
+/// two properties. This allows the color to be animated independently (e.g.
+/// because it is being driven by an [AnimatedTheme]).
+///
+/// The [clipper] is not animated.
+class _AnimatedPhysicalShape extends ImplicitlyAnimatedWidget {
+  /// Creates a widget that animates the properties of a [PhysicalShape] not
+  /// including the [clipper].
+  ///
+  /// The [child], [clipper], [elevation], [color], [shadowColor], [curve], and
+  /// [duration] arguments must not be null.
+  ///
+  /// Animating [color] is optional and is controlled by the [animateColor] flag.
+  ///
+  /// Animating [shadowColor] is optional and is controlled by the [animateShadowColor] flag.
+  const _AnimatedPhysicalShape({
```

Comment:
i'd probably call this _MaterialInterior to avoid confusion with PhysicalShape

Reply:
done